src/libmongoc/tests/test-mongoc-change-stream.c

```diff
@@ -2490,6 +2490,13 @@ prose_test_17 (void)
    request_destroy (request);
 
    /* Resume occurs. */
```

Comment:
The "not primary" error above causes the monitor description to be marked Unknown. During the resume process `mongoc_cursor_destroy` would fail to construct a correct kill cursors command since the server description was Unknown.